### PR TITLE
Adjusts VPA maxAllowed settings for MCM and CSI driver controller

### PR DESCRIPTION
**Pull request by dimitar-kostadinov**
**How to categorize this PR?**

/area auto-scaling
/kind enhancement
/platform gcp

**What this PR does / why we need it**:
Adjusts maxAllowed settings for MCM and CSI driver controller
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: charts/internal/machine-controller-manager/seed/templates/vpa.yaml [hunk 1]
```diff
@@ -18,12 +18,15 @@ spec:
         cpu: {{ .Values.resources.mcmProviderGCP.requests.cpu }}
         memory: {{ .Values.resources.mcmProviderGCP.requests.memory }}
       maxAllowed:
-        cpu: {{.Values.resources.mcmProviderGCP.maxAllowed.cpu}}
-        memory: {{.Values.resources.mcmProviderGCP.maxAllowed.memory}}
+        cpu: 2
+        memory: 5G
```
Comment (kon-angelo) [inline charts/internal/machine-controller-manager/seed/templates/vpa.yaml:22]: I am not sure I like this change very much. Why hardcode the values inside the templates and remove them from the values file ?

**Review by kon-angelo — CHANGES_REQUESTED** [review]

File: charts/internal/machine-controller-manager/seed/templates/vpa.yaml (hunk 1 above)
Comment (dimitar-kostadinov) [reply charts/internal/machine-controller-manager/seed/templates/vpa.yaml:22]: I noticed that the `.Values.resources.*` is serialized in deployment template (e.g. [here](https://github.com/gardener/gardener-extension-provider-gcp/blob/003f0ac79ea5f8b03ea96b901168863ef8a38913/charts/internal/machine-controller-manager/seed/templates/deployment.yaml#L68)), but `maxAllowed` is not part of [resources](https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#resources), so move it directly into the VPA templates

Comment (dimitar-kostadinov) [reply charts/internal/machine-controller-manager/seed/templates/vpa.yaml]: Done

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Pull request merged**